### PR TITLE
Fix MSVC warning wd4146 unary minus operator applied to unsigned type

### PR DESCRIPTION
**Pull request by RossBencina**
Fixes:

> wd4146: W2 unary minus operator applied to unsigned type, result still unsigned

See next comment for details.

### CODE REVIEW

**Comment by RossBencina** [comment]
*CMake build / Windows MSVC (pull_request)* action failed with:

```
 paqa.c
D:\a\portaudio\portaudio\qa\loopback\src\paqa.c(1354,108): error C2220: the following warning is treated as an error [D:\a\portaudio\portaudio\build\qa\loopback\paloopback.vcxproj]
D:\a\portaudio\portaudio\qa\loopback\src\paqa.c(1354,108): warning C4146: unary minus operator applied to unsigned type, result still unsigned [D:\a\portaudio\portaudio\build\qa\loopback\paloopback.vcxproj]
```

**Comment by RossBencina** [comment]
@dechamps just letting you know I think I've fixed this, incase you were tempted to look at it.

File: qa/loopback/src/paqa.c [hunk 1]
```diff
@@ -1351,7 +1351,7 @@ int TestSampleFormatConversion( void )
     const char charInput[] = { 127, 64, -64, -128 };
     const unsigned char ucharInput[] = { 255, 128+64, 64, 0 };
     const short shortInput[] = { 32767, 32768/2, -32768/2, -32768 };
-    const int intInput[] = { 2147483647, 2147483647/2, -1073741824 /*-2147483648/2 doesn't work in msvc*/, -2147483648 };
+    const int intInput[] = { 2147483647, 2147483647/2, /*-2147483648/2:*/(-2147483647 - 1)/2, /*"-2147483648":*/(-2147483647 - 1) }; /*see PR #814*/
```
Comment (RossBencina) [inline qa/loopback/src/paqa.c:1354]: Alternatively we could do something like:
```suggestion
    const int int_minus_2147483648 = (-2147483647 - 1); /*"-2147483648" as a signed integer. See PR #814*/
    const int intInput[] = { 2147483647, 2147483647/2, int_minus_2147483648/2, int_minus_2147483648 }; /*see PR #814*/
```

**Comment by dechamps** [comment]
Is there any reason why we can't change the code to use the standard constants such as `INT_MIN`, `INT_MAX` etc. instead of literal numbers? E.g. something like

```c++
const int intInput[] = { INT_MAX, INT_MAX/2, INT_MIN/2, INT_MIN };
```

This would seem like a cleaner/more readable way to avoid the warning.

File: qa/loopback/src/paqa.c [hunk 2]
```diff
@@ -1351,7 +1351,8 @@ int TestSampleFormatConversion( void )
     const char charInput[] = { 127, 64, -64, -128 };
     const unsigned char ucharInput[] = { 255, 128+64, 64, 0 };
     const short shortInput[] = { 32767, 32768/2, -32768/2, -32768 };
-    const int intInput[] = { 2147483647, 2147483647/2, -1073741824 /*-2147483648/2 doesn't work in msvc*/, -2147483648 };
+    const int int_minus_2147483648 = (-2147483647 - 1); /*"-2147483648" as a signed integer. See PR #814*/
+    const int intInput[] = { 2147483647, 2147483647/2, int_minus_2147483648/2, int_minus_2147483648 }; /*see PR #814*/
```
Comment (RossBencina) [inline qa/loopback/src/paqa.c:1355]:
```suggestion
    const int intInput[] = { 2147483647, 2147483647/2, int_minus_2147483648/2, int_minus_2147483648 };
```

**Review by philburk — APPROVED** [review]

**Comment by RossBencina** [comment]
@dechamps because we want `INT32_MIN`/`INT32_MAX` (i.e. always 32-bit values) irrespective of the size of int. And those symbols are not present in C89.

**Pull request merged**